keps/sig-network/0013-20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -0,0 +1,362 @@
+            * [NGINX Ingress Controller - Dual-Stack Support for Bare Metal Clusters](#nginx-ingress-controller---dual-stack-support-for-bare-metal-clusters)
+         * [Load Balancer Operation ???](#load-balancer-operation-)
+         * [Network Policy Considerations ???](#network-policy-considerations-)
+         * [Cloud Provider Plugins Considerations ???](#cloud-provider-plugins-considerations-)
+         * [kubeadm Support](#kubeadm-support)
+         * [End-to-End Test Support](#end-to-end-test-support)
+         * [User Stories [optional]](#user-stories-optional)
+            * [Story 1](#story-1)
+            * [Story 2](#story-2)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Graduation Criteria](#graduation-criteria)
+      * [Implementation History](#implementation-history)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+         * [Dual Stack at the Edge &lt;more info TBD&gt;](#dual-stack-at-the-edge-more-info-tbd)
+
+## Summary
+
+This proposal adds full IPv4/IPv6 dual stack functionality to Kubernetes clusters. This includes the following concepts:
+- Awareness of multiple IPv4/IPv6 address assignments per pod
+- Support for multiple service IPv4/IPv6 assignments per service
+- Native IPv4-to-IPv4 in parallel with IPv6-to-IPv6 communications to, from, and within a cluster
+
+## Motivation
+
+The adoption of IPv6 has increased in recent years, and customers are requesting IPv6 support in Kubernetes clusters. To this end, the support of IPv6-only clusters was added as an alpha feature in Kubernetes Version 1.9. Clusters can now be run in either IPv4-only, IPv6-only, or in a very limited dual-stack configuration. This limited dual-stack support might be called "headless", in the sense that a CNI network plugin can configure dual-stack addresses on a pod, but Kubernetes remains aware of only a single IP address per pod.  This "headless" dual-stack support is limited by the following restrictions:
+- Some CNI plugins are capable of assigning dual-stack addresses on a pod, but Kubernetes is aware of only one address per pod.
+- Service IPs are either all IPv4 or all IPv6 in a cluster.
+- Kubernetes system pods (api server, controller manager, etc.) can have only one IP address per pod, and addresses are either all IPv4 or all IPv6.
+- Endpoints for services are either all IPv4 or all IPv6.
+- Kube-dns is capable of running dual-stack, but it is currently only made aware of either all IPv4 or all IPv6 service addresses.
+
+For scenarios that require legacy IPv4-only clients or services, the above restrictions mean that complex and expensive IPv4/IPv6 transition mechanisms (e.g. NAT64/DNS64, stateless NAT46, or SIIT/MAP) will need to be implemented in the data center networking.
+
```

Comment:
It would be useful to have a section on the use cases that need to be addressed, IMHO, so that people have a picture of how the proposal(s) would address the needs. I know, it's a big ask...